internal/app/secretless/handlers/mysql/backend.go

```diff
@@ -70,273 +118,310 @@ func (h *Handler) ConfigureBackend() (err error) {
 		result.Options[k] = string(v)
 	}
 
-	h.BackendConfig = &result
+	return result, nil
+}
+
+// https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase.html
+//
+type ConnectionPhase struct {
```

Comment:
I'd love to see this moved into the protocol

Reply:
So the design is: protocol provides the abstractions over the low level protocol details.  This thing (`ConnectionPhase` above but soon to be `AuthenticationHandshake`) uses those abstractions to actually broker the connection.  That is, this code is not _part_ of the protocol, it _uses_ the protocol.  Or again: This code is the heart of the secretless handler.  A secretless handler is  not part of the MySQL protocol.  

So we don't want to move this into protocol.  That said, the `backend` vs `handler` distinction is a bit vague.